packages/components/src/radio-control/stories/index.story.tsx

```diff
@@ -91,3 +92,55 @@ WithOptionDescriptions.args = {
 		},
 	],
 };
+
+export const WithComponentLabels: StoryFn< typeof RadioControl > =
```

Comment:
I would suggest these changes for simplicity:

- Use an icon from the library.
- Add a description for accessibility guidance.
- Use the standard `aria-label` prop.
- Remove the `title` (it's not what we want to highlight here, and we usually use Tooltip for these things).
- Decrease the number of options to the minimum viable number to demonstrate the point.

```diff
diff --git a/packages/components/src/radio-control/stories/index.story.tsx b/packages/components/src/radio-control/stories/index.story.tsx
index 3b3efbe952..108a58b12f 100644
--- a/packages/components/src/radio-control/stories/index.story.tsx
+++ b/packages/components/src/radio-control/stories/index.story.tsx
@@ -7,12 +7,13 @@ import type { Meta, StoryFn } from '@storybook/react';
  * WordPress dependencies
  */
 import { useState } from '@wordpress/element';
+import { starFilled } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import RadioControl from '..';
-import { Path, SVG } from '@wordpress/primitives';
+import Icon from '../../icon';
 
 const meta: Meta< typeof RadioControl > = {
 	component: RadioControl,
@@ -93,27 +94,20 @@ WithOptionDescriptions.args = {
 	],
 };
 
+/**
+ * When the label is not a string, make sure that the element is accessibly labeled.
+ */
 export const WithComponentLabels: StoryFn< typeof RadioControl > =
 	Template.bind( {} );
 
-function Rating( { stars, ariaLabel }: { stars: number; ariaLabel: string } ) {
-	const starPath =
-		'M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z';
+function Rating( {
+	stars,
+	...restProps
+}: { stars: number } & JSX.IntrinsicElements[ 'div' ] ) {
 	return (
-		<div
-			style={ { display: 'flex' } }
-			aria-label={ ariaLabel }
-			title={ ariaLabel }
-		>
+		<div style={ { display: 'flex' } } { ...restProps }>
 			{ Array.from( { length: stars }, ( _, index ) => (
-				<SVG
-					key={ index }
-					width={ 24 }
-					height={ 24 }
-					viewBox="0 0 24 24"
-				>
-					<Path d={ starPath } />
-				</SVG>
+				<Icon key={ index } icon={ starFilled } />
 			) ) }
 		</div>
 	);
@@ -123,23 +117,15 @@ WithComponentLabels.args = {
 	label: 'Rating',
 	options: [
 		{
-			label: <Rating stars={ 5 } ariaLabel="Five Stars" />,
-			value: '5',
-		},
-		{
-			label: <Rating stars={ 4 } ariaLabel="Four Stars" />,
-			value: '4',
-		},
-		{
-			label: <Rating stars={ 3 } ariaLabel="Three Stars" />,
+			label: <Rating stars={ 3 } aria-label="Three Stars" />,
 			value: '3',
 		},
 		{
-			label: <Rating stars={ 2 } ariaLabel="Two Stars" />,
+			label: <Rating stars={ 2 } aria-label="Two Stars" />,
 			value: '2',
 		},
 		{
-			label: <Rating stars={ 1 } ariaLabel="One Star" />,
+			label: <Rating stars={ 1 } aria-label="One Star" />,
 			value: '1',
 		},
 	],
```

Reply:
addressed here b9a85c08d4f79b42b999104658b9396af05428fd

<img width="386" alt="image" src="https://github.com/user-attachments/assets/9d20e653-76b9-4ca7-91ef-186f02a258d6">